### PR TITLE
WER scraper (#42)

### PR DESCRIPTION
**Pull request by void-elf**
Implements #42.

Some notes:

1. There are some overlaps between what we scrape from WER and data from Jack that we manually imported. I think that instead of manual curation by a human to remove overlaps, we should have code somewhere that detects overlaps and doesn't display overlapping results. As more data sources get added, this will save us a lot of time.

2. The [tables for the WER data](https://github.com/syhw/wer_are_we/) sometimes have multiple columns for scores, and I wasn't sure which to pick (or even how to handle multiple scores for what would have basically been the same "measure"), so I just always went with the first one. 

You can see a preview of the data generated in `data/wer.py`, and can run the scraper yourself via `python -m scraper.wer`.

### CODE REVIEW

**Comment by pde** [comment]
For the multiple columns, I think what we're seeing is either different training constraints/data types, or different training and test data types. Compare to bAbi 20 QA, where there are different training setups with either 1K or 10K examples, or to the Atari data where you have either the human starts or the 0-30 no-op starts.

There are multiple ways we could handle this:

1. Make multiple metrics (for instance, `librispeech test-clean` and `librispeech test-other` or drop the word "test"). Similar to what we do with bAbi 20 QA.
2. Keep one metric but indicate on each measure which of the types it is: `HMM-DNN + pNorm* clean` `HMM-DNN + pNorm* other` for instance.
3. Implement some special logic so that the metrics know about these "flavours", and can put them on a single graph with different frontiers of best performance

In the long run 3 is probably the "right" way to do this, but it could be okay to do 1 or 2 for now.  1 has the advantage of keeping each metric an "apples to apples" comparison; 2 has the advantage of reducing the number of graphs in the Notebook.  In this case because some of the variants are systematically easier than others, 1 is probably the way to go.  For Atari HS vs NOOP 2 made sense because it doesn't seem that one of these is systematically easier than the other.

**Review by pde — CHANGES_REQUESTED** [review]
Per my longer comment, we should ingest all of the results columns, either by method 1 (making multiple metrics) or method 3 (implementing something like "flavours" for metrics).

**Comment by void-elf** [comment]
@pde I went with method 1, although I agree that something like method 3 is best in the long run. Let me know what you think!

**Review by pde — APPROVED** [review]

File: scrapers/wer.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+def get_metrics(h2_name, first_row, file_output, scale = "error_percent"):
+    metric_names = []
+    name = h2_name_map[h2_name]
+    if name == "":
+        raise SystemExit("Error! Need to add h2_name to h2_name_map to be able to parse!")
+    for column in first_row:
+        column = str(column.getText()).translate(None, string.punctuation)
+        metric_name = str(name + "_" + "_".join(re.findall(r"[\w']+", column))).replace("test", "")
+        metric_names.append(metric_name)
+        s = "{0} = speech_recognition.metric(name=\"{1} {2}\", scale={3}, attributes=['language'])\n".format(metric_name, name, column, scale)
```
Comment (pde) [inline scrapers/wer.py]: `attributes` is only a parameter for `Problem` objects, not `Metric`s.

File: scrapers/wer.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                'name': columns[l-1].encode('ascii', 'ignore'),
+                'date': date,
+                'url': url,
+                'values': map(lambda x: x.strip('%'), columns[:-3])
+            }
+        return r
+    data = row_data(row)
+    notes = data['name']
+    if notes == "Humans":
+        # TODO: eventually use the human measurements as benchmarks. That's kind of hard right now.
```
Comment (pde) [inline scrapers/wer.py]: It's easy to do! Just pass them to the `.metric` call as the `target` parameter. You can also pass the `WER are we` url as `target_source` where there's a human performance number available.

**Review by pde — CHANGES_REQUESTED** [review]
Looks good, and close to mergeable. Though it seems you didn't ever get the `wer.py` data to import?

File: scrapers/wer.py (hunk 2 above)
Comment (void-elf) [reply scrapers/wer.py]: Yes, I know that it's easy to add a `target` parameter to `.metric` call, but this comment was referencing a different issue. Implementing this right requires a bit of restructuring, because by the time those targets are read in the lower half of the table, the metrics have already been written. I'll fix this soon.

**Comment by void-elf** [comment]
To get the WER data to import, this is all that needs to be added to the notebook under the speech recognition section:

```
for wer_metric in wer_metrics:
    if wer_metric.name != "Switchboard Hub5'00":
        wer_metric.graph()
```
Something really weird is going on with my copy of ipnb dropout, and so I can't commit that small change without it wanting to include lots of other stuff. Could you add it in? Using that small code snippet, the new graphs + tables are displaying nicely for me.

Also, I think I've addressed all your feedback! Let me know if there's anything else I should change :)
